src/openfl/_internal/text/TextEngine.hx

```diff
@@ -213,16 +213,24 @@ class TextEngine {
 		}
 	}
 
-	public function getLineBreakIndex(startIndex:Int = 0):Int {
+	function getWordSeparatorIndex(startIndex:Int):Int {
+		var space = text.indexOf(" ", startIndex);
+		var hyphen = text.indexOf("-", startIndex);
+		return
+			if (space == -1) hyphen
+			else if (hyphen == -1) space
+			else if (space < hyphen) space
+			else hyphen;
+	}
+
+	public function getLineBreakIndex(startIndex:Int):Int {
 		var cr = text.indexOf("\n", startIndex);
 		var lf = text.indexOf("\r", startIndex);
```

Comment:
btw. as I see it now: CR should actually be "\r" and LF should be "\n", but that's quite minor ;)